lib/spack/spack/test/compilers.py

```diff
@@ -79,3 +80,174 @@ def test_compiler_flags_from_config_are_grouped():
 
     compiler = compilers.compiler_from_config_entry(compiler_entry)
     assert any(x == '-foo-flag foo-val' for x in compiler.flags['cflags'])
+
+
+# Test behavior of flags and UnsupportedCompilerFlag.
+
+# Utility function to test most flags.
+default_compiler_entry = {
+    'spec': 'clang@2.0.0-apple',
+    'operating_system': 'foo-os',
+    'paths': {
+        'cc': 'cc-path',
+        'cxx': 'cxx-path',
+        'fc': None,
+        'f77': None
+    },
+    'flags': {},
+    'modules': None
+}
+
+
+# Fake up a mock compiler where everything is defaulted.
+class MockCompiler(Compiler):
+    def __init__(self):
+        super(MockCompiler, self).__init__(
+            "badcompiler@1.0.0",
+            default_compiler_entry['operating_system'],
+            None,
+            [default_compiler_entry['paths']['cc'],
+             default_compiler_entry['paths']['cxx'],
+             default_compiler_entry['paths']['fc'],
+             default_compiler_entry['paths']['f77']])
+
+    @property
+    def name(self):
+        return "mockcompiler"
+
+    @property
+    def version(self):
+        return "1.0.0"
+
+
+# Get the desired flag from the specified compiler spec.
+def flag_value(flag, spec):
+    compiler = None
+    if spec is None:
+        compiler = MockCompiler()
+    else:
+        compiler_entry = copy(default_compiler_entry)
+        compiler_entry['spec'] = spec
+        # Disable faulty id()-based cache (issue #7647).
+        compilers._compiler_cache = {}
+        # Note that compiler is used within an eval() statement below
+        # but not otherwise, so we need to ignore flake8 error F841.
+        # Unfortunately the error is tagged to the line with the = on
+        # it, so we cannot continue the line and must therefore also
+        # deactivate the long line error, E501.
+        compiler = compilers.compiler_from_config_entry(compiler_entry)  # noqa: F841, E501
+
+    return eval("compiler.{0}".format(flag))
```

Comment:
@chissg can you use `getattr` to avoid `eval` (and the above comment/noqa)?

Reply:
@scheibelp Works on Python 2, at least: uploading for tests now. Thanks for the suggestion, I wouldn't have thought that would work on `@property`-tagged functions, but it does.